components/infra-proxy-service/storage/postgres/migrations.go

```diff
@@ -143,3 +142,18 @@ func (p *postgres) insertMigration(ctx context.Context, migrationId, serverId, m
 
 	return m, nil
 }
+
+//GetActiveMigration gets the Migration ID and Migration Status for a server id
+func (p *postgres) GetActiveMigration(ctx context.Context, serverId string) (storage.ActiveMigration, error) {
+	var m storage.ActiveMigration
+	query := "select m.id,t.type from migration m join migration_type t on m.type_id=t.id and t.id < 5000 and m.server_id=$1;"
```

Comment:
We need to make sure that we are picking up the latest status and the migration id. We can make sure that only one migration occurs for a server but we should be fetching the latest status ordered by the updated timestamp.